tests/generate/test_generate_sets.py

```diff
@@ -59,3 +61,65 @@ def test_make_sets_from_sampler():
     # Test with non-positive number of runs
     result = make_sets_from_sampler(mock_sampler, num_runs=0)
     assert result == []
+
+
+# Test case 1: Ensure unique sets are generated
+def test_make_sets_generates_unique_sets():
+    # Set up some sample hyperparameters
+    hps = {
+        "set_types": ["numbers"],
+        "n": 20,
+        "m_A": 5,
+        "m_B": 5,
+    }
+
+    # Generate sets using make_sets
+    sets = make_sets(**hps, number_of_data_points=10, seed_value=42)
+
+    # Check that all sets are unique
+    n_duplicated_A = pd.DataFrame(sets)["A"].duplicated().sum()
```

Comment:
couldn't make it work last time, because of the data type issue (set of tuples)

i'm converting it to string and then do the set generation to reduce the pandas dependency now. i'll push the PR shortly

Reply:
I think you want this:
```
sets = {"A": ((2, 3), (2, 3), (8, 9)), "B": ((5, 7), (5, 7), (2, 4))}
# Make a set of (a, b) tuples, where a and b are individual sets.
set_pairs = {(a, b) for a, b in zip(sets["A"], sets["B"])} 
# Verify that all of these (a, b) pairs are unique.
assert len(set_pairs) == len(sets["A"]) == len(sets["B"])
```

Reply:
Yeah, the issue is that the formatting of the sets is not like a dictionary for "A" and "B".  It's a list like:
`[{"A": (2,3), "B": (5,7)}, {"A": (2,3), "B": (5,7)}, {"A": (8,9), "B": (2,4)}]`

I modified the function to be granular and make sure we can catch a duplicate pair, like this:
```
sets = make_sets(**hps, number_of_data_points=1000, seed_value=42)
pairs = set()
for d in sets:
    a = tuple(sorted(d["A"]))
    b = tuple(sorted(d["B"]))
    pair = (a, b)
    assert pair not in pairs, f"Duplicate pair found: A={a}, B={b}"
    pairs.add(pair)
 ```